### PR TITLE
Upgrade to SynapseAI 1.19 version

### PR DESCRIPTION
**Pull request by kaixuanliu**

### CODE REVIEW

**Comment by kaixuanliu** [comment]
We made a benchmark to compare the perf with 1.18, using 5 models listed below:
- BAAI/bge-large-en-v1.5
- sentence-transformers/all-mpnet-base-v2
- sentence-transformers/all-MiniLM-L6-v2
- sentence-transformers/paraphrase-albert-small-v2
- intfloat/e5-mistral-7b-instruct

Best throughput for each model are:
|       | 1.18 | 1.19
| ----------- | ----------- | ----------- |
| BAAI/bge-large-en-v1.5      | 1521       | 1370 |
| sentence-transformers/all-mpnet-base-v2   |    2662     | 2698 |
| sentence-transformers/all-MiniLM-L6-v2   | 5175        | 4539 |
| sentence-transformers/paraphrase-albert-small-v2   | 3251        | 3181 |
| intfloat/e5-mistral-7b-instruct   | 153        | 156 |

basically aligned with 1.18, except 2 larger perf drop for `BAAI/bge-large-en-v1.5` and `sentence-transformers/all-MiniLM-L6-v2`, this is due to the Pytorch 2.5 Performance Degradation issue mentioned in [Optimum-habana](https://github.com/huggingface/optimum-habana?tab=readme-ov-file#important-note-on-pytorch-25-performance-degradation), we added `torch._C._set_math_sdp_allow_fp16_bf16_reduction(True)` as a WA(The data above is w/ this WA, otherwise will be even worse).

**Comment by kaixuanliu** [comment]
@regisss , pls help review

**Review by regisss — APPROVED** [review]
LGTM!
